### PR TITLE
[TIMOB-18162] Autodetect proxy configuration within ti setup

### PR DESCRIPTION
**Pull request by skypanther**
https://jira.appcelerator.org/browse/TIMOB-18162

### CODE REVIEW

File: lib/commands/setup.js [hunk 1]
```diff
@@ -1253,9 +1260,22 @@ SetupScreens.prototype.app = function app(callback) {
  * @param {Function} callback - Function to be called when the prompting finishes
  */
 SetupScreens.prototype.network = function app(callback) {
+	var defaultProxy = '';
+	if(this._config.get('cli.httpProxyServer')) {
+		defaultProxy = this._config.get('cli.httpProxyServer');
+	} else if(proxy.length > 0) {
+		var pI = 0, pL = proxy.length;
+		for(; pI < pL; pI++) {
```
Comment (sujmishra) [inline lib/commands/setup.js]: can be just :
```Javascript
for(pI = 0; pI < proxy.length; pI++) {
```

File: lib/commands/lib/proxy_detect.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+					callback(parseNetSetup(stdout));
+				}
+			});
+		})
+	} else {
+		if (process.env['https_proxy'] != undefined) {
+			// if both configured, https proxy is preferentially returned
+			callback(parseEnv(process.env['https_proxy']));
+		} else if (process.env['http_proxy'] != undefined) {
+			callback(parseEnv(process.env['https_proxy']));
```
Comment (sujmishra) [inline lib/commands/lib/proxy_detect.js]: if http_proxy is configured should the callback be to http_proxy?  (in the else if)

File: lib/commands/setup.js (hunk 1 above)
Comment (skypanther) [inline lib/commands/setup.js]: except that then pI and pL would be global variables. I admit the form is a bit unusual but it follows the pattern of other code in the project.

File: lib/commands/lib/proxy_detect.js (hunk 2 above)
Comment (skypanther) [inline lib/commands/lib/proxy_detect.js]: yes, thanks for that good catch. I'll fix it.

File: lib/commands/setup.js [hunk 3]
```diff
@@ -1253,9 +1260,22 @@ SetupScreens.prototype.app = function app(callback) {
  * @param {Function} callback - Function to be called when the prompting finishes
  */
 SetupScreens.prototype.network = function app(callback) {
+	var defaultProxy = '';
+	if(this._config.get('cli.httpProxyServer')) {
```
Comment (cb1kenobi) [inline lib/commands/setup.js]: Should be a space between `if` and `(`.

File: lib/commands/setup.js (hunk 1 above)
Comment (cb1kenobi) [inline lib/commands/setup.js]: Typically I would just do:

~~~
for (var i = 0, len = proxy.length; i++) {
~~~

`pI` and `pL` are horrible names.

If you're feeling functional, you could do:

~~~
proxy.some(function (p) {
    if (p && p.valid) {
        defaultProxy = p.fullAddress;
        console.log(defaultProxy);
        return true;
    }
});
~~~

**Comment by sujmishra** [comment]
merging..

**Pull request merged**